### PR TITLE
Refactors PeerConnector and TorrentManager

### PR DESCRIPTION
**Pull request by Lorraine-Lyu**
To make call structure cleaner.

### CODE REVIEW

File: owltorrent-network/src/main/java/edu/rice/owltorrent/network/TorrentManager.java
Comment (JoshuaEng) [inline owltorrent-network/src/main/java/edu/rice/owltorrent/network/TorrentManager.java:362]: Is it possible in our code for the piece length not to be a multiple of block size? If so this will be wrong for the last block in a given piece.

File: owltorrent-network/src/main/java/edu/rice/owltorrent/network/MessageHandler.java
Comment (JoshuaEng) [inline owltorrent-network/src/main/java/edu/rice/owltorrent/network/MessageHandler.java:13]: why does this have a removePeer method but no addPeer method? do we need the removePeer at all?

File: owltorrent-network/src/main/java/edu/rice/owltorrent/network/TorrentManager.java
Comment (JoshuaEng) [inline owltorrent-network/src/main/java/edu/rice/owltorrent/network/TorrentManager.java:412]: Does this make more sense as a method called handleMessage?

**Review by JoshuaEng — APPROVED** [review]
Seems reasonable, I have a couple of small comments but I'm approving so I don't block merge. I didn't give it a super thorough overview but it looks pretty good

File: owltorrent-network/src/main/java/edu/rice/owltorrent/network/MessageHandler.java
Comment (maxyu1115) [reply owltorrent-network/src/main/java/edu/rice/owltorrent/network/MessageHandler.java:13]: Remove peer is a desperate attempt at adding a shutdown hook. Basically when the connection is broken, we want to close the peer connector and remove the peer from the torrent manager. But we don't want to pass the torrent manager into the peer connector

File: owltorrent-network/src/main/java/edu/rice/owltorrent/network/TorrentManager.java
Comment (maxyu1115) [reply owltorrent-network/src/main/java/edu/rice/owltorrent/network/TorrentManager.java:412]: So this was done to have PeerConnector no longer depend on TorrentManager. Which makes no sense from a design standpoint, but was done to save time... (Making it a method we'd still need to make a messageHandler to pass into peerConnectors, or we need to pass in TorrentManager)

File: owltorrent-network/src/main/java/edu/rice/owltorrent/network/TorrentManager.java
Comment (maxyu1115) [reply owltorrent-network/src/main/java/edu/rice/owltorrent/network/TorrentManager.java:362]: So if it is the last block and the last piece length is a multiple of block size, that means the last block size is equal to the default block size.

**Pull request merged**
